### PR TITLE
Cloud authentication AAD B2C

### PR DESCRIPTION
**Pull request by damienbod**
fixes https://github.com/dotnet/AspNetCore.Docs/issues/27793

### CODE REVIEW

**Comment by damienbod** [comment]
@JandosKh I created a PR for this now. Points 1-3 are fixed. I removed the AAD auth demo and only have docs for user flows which is the way to authenticate when using B2C. 

Point 4 TenantId is required for Microsoft.Identity.Web

Point 5 is Azure App registration docs for B2C, I think we should just link this in the docs. Opinions?

@Rick-Anderson @CamSoper 

Tested this using .NET 7 and updated the demo snippets.

**Comment by JandosKh** [comment]
Thank you for PR.
Regarding point 5, I think it will suffice if the doc contains a link to
Azure App registration docs for B2C.

File: aspnetcore/security/authentication/azure-ad-b2c.md [hunk 1]
```diff
@@ -106,18 +101,21 @@ In this tutorial, you'll learn how to configure an ASP.NET Core app for authenti
...
-        :::code language="csharp" source="azure-ad-b2c/sample/Startup.cs" id="snippet_Configure" highlight="19":::
+        :::code language="csharp" source="azure-ad-b2c/sample/Program.cs" id="snippet_app":::
 
         The preceding code enables authentication in ASP.NET Core.
 
 ## Run the app
 
+    > [!NOTE]
+    > Use the profile which matches the Azure App registration **Redirect URIs**
+
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/azure-ad-b2c.md:114]:
```suggestion
> [!NOTE]
> Use the profile which matches the Azure App registration **Redirect URIs**
```

**Comment by Rick-Anderson** [comment]
@damienbod  I did a commit to fix the build so you'll have to pull before you push to prevent conflicts.

**Comment by damienbod** [comment]
I get all my ASP.NET Core Azure AD and Azure AD B2C docs from here:

https://github.com/AzureAD/microsoft-identity-web

@Rick-Anderson 

Would that be enough? Or should we cross ref. other docs?

**Comment by Rick-Anderson** [comment]
@CamSoper please review or request another reviewer.

File: aspnetcore/security/authentication/azure-ad-b2c.md [hunk 2]
```diff
@@ -81,23 +76,23 @@ In this tutorial, you'll learn how to configure an ASP.NET Core app for authenti
     - Renders a **Sign out** or **Sign in** link as appropriate.
         - The link points to an action method on the `Account` controller in the `MicrosoftIdentity` area.
 
-1. In `Views/Shared/_Layout.cshtml`, add the highlighted line within the `<header>` element:
+1. In `Pages/Shared/_Layout.cshtml`, add the highlighted line within the `<header>` element:
```
Comment (CamSoper) [inline aspnetcore/security/authentication/azure-ad-b2c.md:79]:
```suggestion
1. In *Pages/Shared/_Layout.cshtml*, add the highlighted line within the `<header>` element:
```

File: aspnetcore/security/authentication/azure-ad-b2c.md [hunk 3]
```diff
@@ -81,23 +76,23 @@ In this tutorial, you'll learn how to configure an ASP.NET Core app for authenti
...
 
-    :::code language="razor" source="azure-ad-b2c/sample/Pages/Shared/_Layout.cshtml" range="11-32" highlight="10":::
+    :::code language="razor" source="azure-ad-b2c/sample/Pages/Shared/_Layout.cshtml" range="12-33" highlight="18":::
  
     Adding `<partial name="_LoginPartial" />` renders the `_LoginPartial.cshtml` partial view in every page request that uses this layout.
 
-1. In `Startup.cs`, make the following changes:
+1. In `Program.cs`, make the following changes:
```
Comment (CamSoper) [inline aspnetcore/security/authentication/azure-ad-b2c.md:85]:
```suggestion
1. In *Program.cs*, make the following changes:
```

File: aspnetcore/security/authentication/azure-ad-b2c.md [hunk 4]
```diff
@@ -81,23 +76,23 @@ In this tutorial, you'll learn how to configure an ASP.NET Core app for authenti
...
 
     1. Add the following `using` directives:
     
-        :::code language="csharp" source="azure-ad-b2c/sample/Startup.cs" id="snippet_NewUsings":::
+        :::code language="csharp" source="azure-ad-b2c/sample/Program.cs" id="snippet_NewUsings":::
 
         The preceding code resolves references used in the next steps.
 
-    1. Replace `ConfigureServices` with the following code:
+    1. Update the `builder services` with the following code:
```
Comment (CamSoper) [inline aspnetcore/security/authentication/azure-ad-b2c.md:93]:
```suggestion
    1. Update the `builder. Services` lines with the following code:
```

**Review by CamSoper — CHANGES_REQUESTED** [review]
LGTM! Couple of small changes. Two for formatting, one for phrasing.

**Comment by CamSoper** [comment]
Oh and @damienbod I cannot thank you enough for doing this!

File: aspnetcore/security/authentication/azure-ad-b2c.md (hunk 2 above)
Comment (damienbod) [reply aspnetcore/security/authentication/azure-ad-b2c.md:79]: fixed

File: aspnetcore/security/authentication/azure-ad-b2c.md (hunk 3 above)
Comment (damienbod) [reply aspnetcore/security/authentication/azure-ad-b2c.md:85]: fixed

File: aspnetcore/security/authentication/azure-ad-b2c.md (hunk 4 above)
Comment (damienbod) [reply aspnetcore/security/authentication/azure-ad-b2c.md:93]: fixed

**Comment by damienbod** [comment]
@CamSoper Thanks for the review and the feedback. Updated to PR.

**Review by CamSoper — APPROVED** [review]

**Pull request merged**